sync/src/types/mod.rs

```diff
@@ -1506,11 +1506,19 @@ impl SyncShared {
                 }
             },
         );
-        self.state.header_map.insert(header_view.clone());
+
         self.state
             .peers()
             .may_set_best_known_header(peer, header_view.as_header_index());
-        self.state.may_set_shared_best_header(header_view);
+        // This judgment assumes that the header processing is continuous:
+        //
+        // 1. The header process ensures that header processing is continuous.
+        // 2. When a block jump(can't find its parents) occurs, the compact block
+        //    will skip the processing of the block and use the get header message first.
+        // 3. If there is no block skipping, compact block processing is also sequential
+        if self.state.may_set_shared_best_header(header_view.clone()) {
+            self.state.header_map.insert(header_view);
+        }
```

Comment:
suggest add info level log in else.

Reply:
According to the ckb block difficulty algorithm, any uncle block will trigger this else branch